test/nested/apps_1175_outs_not_uploaded_outer_results.json

```diff
@@ -0,0 +1,4 @@
+{
+  "apps_1175_outs_not_uploaded_outer.stage-1.ni___test_inner1_stage_0_integration_check": "It works: upload_intermediateupload_intermediate",
+  "apps_1175_outs_not_uploaded_outer.apps_1175_outs_not_uploaded_outer_frag_stage_0_stage_1_ni___test_inner1_stage_0_integration_check": "It works: upload_intermediateupload_intermediate"
```

Comment:
Have the output names always been so long (such multiple IDs concatenation)?

Reply:
Nope, this is introduced with the new functionality, but only for intermediate outputs. The reason for that is to make them unique so that there are no conflicts if a name repeats across multiple namespaces (e.g. task t1 can have `results` output and task t2 can have `results`). If there are duplicates, there will be an applet compilation error (platform behavior). So to differentiate those outputs, I'm adding a namespace that includes stage name and stage id.

Reply:
Got it, makes sense.